Cpp_test/lib/ipc-queue-send.cpp

```diff
@@ -0,0 +1,97 @@
+#include "ipc-queue-send.h"
+#include <ctime>
+#include <iostream>
+#include <unistd.h>
+
+ipcQueueSend::~ipcQueueSend()
+{
+	mq_close(this->mqd);
+	mq_unlink(this->info.method_name);
+}
+
+void ipcQueueSend::init()
+{
+	// remove old queue name (if any)
+	mq_unlink(this->info.method_name);
+	errno = 0; // clear errno
+	this->mqd = mq_open(this->info.method_name, O_CREAT | O_EXCL | O_WRONLY , 0660, &(this->attr));
+
+	if (this->mqd == -1){
+		throw std::runtime_error(std::string("ERROR: open queue send name ") + this->info.method_name );
+	}
+	else{
+		std::cout << this->info.method_name << " is opened." << std::endl;
+	}
+	this->file_handler.open_file();
+}
+
+void ipcQueueSend::transfer()
+{
+	
+	struct timespec	ts;
+	std::cout << "transfer\n";
+	long mq_send_return_value{0};
+	long read_bytes{0};
+	std::vector<char> buffer(this->attr.mq_msgsize);
+	unsigned total_sent_bytes{0};
+	unsigned long file_size = this->file_handler.get_file_size();
+	if (file_size == 0){
+		throw std::runtime_error("ERROR: File size = 0.");
+	}
+		
+	std::cout << "Sending..." << std::endl;
+	std::cout << "file_size: " << file_size << std::endl;
+	
+	while (total_sent_bytes < file_size)
```

Comment:
You are comparing a unsigned with an unsigned long. This is bad practice and won't work for files more than 4.2Gb

Reply:
Maybe you could check read_bytes instead, and after the while verify that total_sent_bytes == file_size ?

Reply:
I will change both of them to unsigned long long